src/pymor/parameters/base.py

```diff
@@ -616,12 +622,18 @@ def __init__(self, parameters, *ranges):
                    and all(isinstance(v, Number) for v in ranges[k])
                    and ranges[k][0] <= ranges[k][1]
                    for k in parameters)
-        self.parameters = parameters
+        assert constraints is None or callable(constraints)
+        self.__auto_init(locals())
         self.ranges = SortedFrozenDict((k, tuple(v)) for k, v in ranges.items())
 
     def sample_uniformly(self, counts):
         """Uniformly sample |parameter values| from the space.
 
+        In the case of additional :attr:`~ParameterSpace.constraints`, the samples
+        are generated w.r.t. the box constraints specified by
```

Comment:
The constraints might not only be "box" constraints, right?

Reply:
I think either you didn't read the sentence completely, or I don't understand what you are saying ..

Reply:
Of course I did not read the sentence correctly, sorry.